src/carts/CommerceToolsCartPayment.js

```
@@ -124,6 +127,31 @@ class CommerceToolsCartPayment extends CommerceToolsCart {
         });
     }
 
+     /**
+     * Deletes one of the payments from cart and from commerce tools payments.
+     *
+     * @param id                cart id
+     * @param paymentId         the id of the payment
+     * @return {Promise}
+     */
+    deleteCartPayment(id, paymentId) {
+        let ccifId = new CcifIdentifier(id);
+        this.requestBuilder.byId(ccifId.getCommerceToolsId());
+        const baseUrl = this._buildBaseUrl();
+        return this._getCartPaymentIdentifier(baseUrl, paymentId).then(payment => {
+            const data = {
+                actions: [{action: 'removePayment', payment: {id: payment.id, version: payment.version}}]
+            };
+            data.version = this.cartVersion;
+            return this._handle(baseUrl, 'POST', data).then(cart => {
+                this.paymentClient.delete(payment.id, payment.version);
```

This returns a promise. The way you implemented it, you always return the cart, independently of whether the payment method was deleted or not. IMO you should wait for the promise to be resolved and in the case of rejection, we need to return a proper error message.

---

Doesn't `this.paymentClient.delete()` return a resolved promise?! At least this is what I get from https://github.com/adobe/commerce-cif-commercetools/blob/master/src/common/CommerceToolsClientBase.js#L116

---

@dplaton A resolved promise is still a promise that has to be handled like a promise. `Promise.resolve()` is basically a shortcut for 
```
return new Promise((resolve, reject) => {
    resolve();
});
```

---

Funny thing, just noticed that this code has been like this since... forever (I kept wondering my the UT fails after changing the code to wait for the promise to be resolved).